ui/app/machines/evaluations.js

```diff
@@ -0,0 +1,123 @@
+import { assign, createMachine, send } from 'xstate';
+
+export default createMachine(
```

Comment:
Pretty cool pattern! I'd love to see a brief comment on why we're using a state machine here, and info on any conventions.